### DISCUSSION TITLE
Varying a data-generation parameter during SBC

### DISCUSSION THREAD

**Discussion by mike-lawrence** [wlandau/stantargets#29] [Help] [answered]
Say my `generate_data()` function takes an argument `N` specifying the number of data points to generate. Is there a way to specify a `tar_stan_mcmc_rep_summary` target that iterates over a list of values for N?

**Comment by wlandau** [comment, marked as answer]
If I understand correctly, you want to generate multiple datasets with different values of `N` and have `tar_stan_mcmc_rep_summary()` output for both, right? I believe you can accomplish this with static branching via `tarchetypes::tar_map()`. Sketch:

```r
# _targets.R
library(targets)
library(tarchetypes)
library(stantargets)
list(
  tar_map(
    values = list(N = c(100, 500, 1000),
    tar_stan_mcmc_rep_summary(x, data = generate_data(N = N), ...)
  )
)
```

**Reply by mike-lawrence** [reply]
That's precisely it, thanks!
